src/compiler/factory.ts

```diff
@@ -2503,8 +2503,8 @@ namespace ts {
             helpers
         } = sourceEmitNode;
         if (!destEmitNode) destEmitNode = {};
-        if (leadingComments) destEmitNode.leadingComments = addRange(leadingComments.slice(), destEmitNode.leadingComments);
-        if (trailingComments) destEmitNode.trailingComments = addRange(trailingComments.slice(), destEmitNode.trailingComments);
+        if (leadingComments) destEmitNode.leadingComments = concatenate(leadingComments, destEmitNode.leadingComments);

```

Comment:
The reason I used `slice` was I always wanted to be sure `leadingComments` and `trailingComments` were unique arrays on an emit node, in case they are modified later. Now it is possible for someone to change the leading/trailing comments of a cloned node and it will affect the leading/trailing comments of the source node (though its not a likely scenario).